kbase-extension/static/kbase/js/common/jobCommChannel.js

```diff
@@ -261,17 +310,56 @@ define([
 
                 // job information for one or more jobs
                 case RESPONSES.INFO:
+                    Object.keys(msgData).forEach((jobId) => {
+                        // get the job mapping if it doesn't exist
+                        if (!this._jobMapping[jobId]) {
+                            if (msgData[jobId].batch_id) {
+                                this._jobMapping[jobId] = {
+                                    [jcm.PARAM.BATCH_ID]: msgData[jobId].batch_id,
+                                };
+                            } else {
+                                this._jobMapping[jobId] = { [jcm.PARAM.JOB_ID]: jobId };
+                            }
+                        }
+                        this.sendBusMessage(CHANNEL.JOB, jobId, msgType, msgData[jobId]);
+                    });
+                    break;
+
                 case RESPONSES.LOGS:
+                    Object.keys(msgData).forEach((jobId) => {
+                        this.sendBusMessage(CHANNEL.JOB, jobId, msgType, msgData[jobId]);
+                    });
+                    break;
+
                 case RESPONSES.RETRY:
+                    Object.keys(msgData).forEach((jobId) => {
```

Comment:
retry messages should have the batch info for the new retry job, and can also be used to get the retry parent's batch info if it doesn't already exist.